apisix/runner/http/request.py

```diff
@@ -0,0 +1,224 @@
+        self._req_method = ""
+        self._req_path = ""
+        self._req_headers = {}
+        self._req_configs = {}
+        self._req_args = {}
+        self._req_src_ip = ""
+        self._init()
+
+    @property
+    def rpc_type(self) -> int:
+        return self._rpc_type
+
+    @rpc_type.setter
+    def rpc_type(self, rpc_type: int) -> None:
+        self._rpc_type = rpc_type
+
+    @property
+    def rpc_buf(self) -> bytes:
+        return self._rpc_buf
+
+    @rpc_buf.setter
+    def rpc_buf(self, rpc_buf: bytes) -> None:
+        self._rpc_buf = rpc_buf
+
+    @property
+    def conf_token(self) -> int:
+        return self._req_conf_token
+
+    @conf_token.setter
+    def conf_token(self, req_conf_token: int) -> None:
+        self._req_conf_token = req_conf_token
+
+    @property
+    def id(self) -> int:
+        return self._req_id
+
+    @id.setter
+    def id(self, req_id: int) -> None:
+        self._req_id = req_id
+
+    @property
+    def method(self) -> str:
+        return self._req_method
+
+    @method.setter
+    def method(self, req_method: str) -> None:
+        self._req_method = req_method
+
+    @property
+    def path(self) -> str:
+        return self._req_path
+
+    @path.setter
+    def path(self, req_path: str) -> None:
+        self._req_path = req_path
+
+    @property
+    def headers(self) -> dict:
+        return self._req_headers
+
+    @headers.setter
+    def headers(self, req_headers: dict) -> None:
+        self._req_headers = req_headers
+
+    @property
+    def configs(self) -> dict:
+        return self._req_configs
+
+    @configs.setter
+    def configs(self, req_configs: dict) -> None:
+        self._req_configs = req_configs
+
+    @property
+    def args(self) -> dict:
+        return self._req_args
+
+    @args.setter
+    def args(self, req_args: dict) -> None:
+        self._req_args = req_args
+
+    @property
+    def src_ip(self) -> str:
+        return self._req_src_ip
+
+    @src_ip.setter
+    def src_ip(self, req_src_ip: str) -> None:
+        self._req_src_ip = req_src_ip
+
+    def reset(self) -> None:
+        """
+        reset request class
+        :return:
+        """
+        self._rpc_type = 0
+        self._rpc_buf = b''
+        self._req_id = 0
+        self._req_conf_token = 0
+        self._req_method = ""
+        self._req_path = ""
+        self._req_headers = {}
+        self._req_configs = {}
+        self._req_args = {}
+        self._req_src_ip = ""
+
+    def _parse_src_ip(self, req: A6HTTPReqCallReq) -> None:
+        """
+        parse request source ip address
+        :param req:
+        :return:
+        """
+        if not req.SrcIpIsNone():
+            delimiter = "."
+            if req.SrcIpLength() > 4:
+                delimiter = ":"
+            ipAddress = []
+            for i in range(req.SrcIpLength()):
+                ipAddress.append(str(req.SrcIp(i)))
+            self.src_ip = delimiter.join(ipAddress)
```

Comment:
IPv6 isn't `127:127:127...`. It would be better to call a battlefield-test library to do the job for you.

Reply:
fixed